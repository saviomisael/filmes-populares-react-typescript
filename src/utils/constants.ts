export const IMAGE_URL = process.env.REACT_APP_IMAGE_URL;
