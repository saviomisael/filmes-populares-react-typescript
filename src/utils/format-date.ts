export const formatDate = (date: Date) =>
  date.toLocaleString('pt-BR', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });
