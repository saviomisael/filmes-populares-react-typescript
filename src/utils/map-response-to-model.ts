import { TvSerie } from 'models/TvSerie';
import { TvSerieResponse } from 'services/responses/TvSerieResponse';

const mapSingleResponseToTvSerieModel = (response: TvSerieResponse) =>
  new TvSerie(response.id, response.name, response.vote_average, response.poster_path);

export const mapResponseToTvSerieModel = (response: TvSerieResponse[]) =>
  response.map((x) => mapSingleResponseToTvSerieModel(x));
