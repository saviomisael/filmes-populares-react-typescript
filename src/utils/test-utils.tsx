import { render } from '@testing-library/react';
import { MemoryRouter, Switch, Route } from 'react-router-dom';
import { BaseStyles } from 'styles/base';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/theme';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { storeMock } from 'mocks/configure-store-mock';

type optionsWithRouter = {
  initialRoutes?: string[];
  pathname: string;
};

export const renderTheme = (children: ReactNode) =>
  render(
    <ThemeProvider theme={theme}>
      <BaseStyles />
      {children}
    </ThemeProvider>,
  );

export const renderWithStoreProvider = (children: ReactNode) =>
  render(
    <Provider store={storeMock}>
      <ThemeProvider theme={theme}>
        <BaseStyles />
        {children}
      </ThemeProvider>
    </Provider>,
  );

export const renderWithRouter = (children: ReactNode, options: optionsWithRouter) => {
  const initialRoutes = options.initialRoutes && options.initialRoutes.length > 0 ? options.initialRoutes : ['/'];

  return render(
    <Provider store={storeMock}>
      <ThemeProvider theme={theme}>
        <MemoryRouter initialEntries={initialRoutes}>
          <Switch>
            <Route path={options.pathname} render={() => children} />
          </Switch>
        </MemoryRouter>
      </ThemeProvider>
    </Provider>,
  );
};
