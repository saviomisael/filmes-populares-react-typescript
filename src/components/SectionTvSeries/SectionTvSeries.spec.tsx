import { renderWithStoreProvider } from 'utils/test-utils';
import { SectionTvSeries } from '.';

describe('<SectionTvSeries />', () => {
  it('should matches with snapshot', () => {
    const { container } = renderWithStoreProvider(<SectionTvSeries />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
