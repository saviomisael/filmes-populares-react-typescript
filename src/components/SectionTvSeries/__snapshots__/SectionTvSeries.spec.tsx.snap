// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionTvSeries /> should matches with snapshot 1`] = `
.c1 {
  color: #413B3B;
  text-align: left;
  text-transform: none;
  padding-bottom: 2rem;
}

.c0 {
  padding: 1.5rem;
  background: #FA9F42;
}

.c8 {
  font-weight: 400;
  font-size: 1.8rem;
  color: #FFFFFF;
  text-align: center;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-column-gap: 1.4rem;
  column-gap: 1.4rem;
  color: #FFFFFF;
}

.c9::before {
  content: url(star.svg);
}

.c3 {
  width: 24rem;
}

.c7 {
  width: 100%;
  position: absolute;
  right: 0;
  bottom: 0.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1.5rem 0;
  visibility: hidden;
  opacity: 0;
  -webkit-transition: visibility 0.3s,opacity 0.3s;
  transition: visibility 0.3s,opacity 0.3s;
  border-bottom-left-radius: inherit;
  border-bottom-right-radius: inherit;
  background: #413B3B88;
}

.c4 {
  border-radius: 1.2rem;
  position: relative;
}

.c4:hover .c6 {
  visibility: visible;
  opacity: 1;
}

.c5 {
  width: 100%;
  border-radius: inherit;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

<section
  class="c0"
>
  <h1
    class="c1"
  >
    Séries mais assistidas
  </h1>
  <ul
    class="c2"
    data-testid="tvserielistcontainer"
  >
    <li>
      <article
        class="c3"
      >
        <figure
          class="c4"
        >
          <img
            alt="poster da série Round 6"
            class="c5"
            src="https://image.tmdb.org/t/p/original/6gcHdboppvplmBWxvROc96NJnmm.jpg"
          />
          <figcaption
            class="c6 c7"
          >
            <p
              class="c8"
            >
              Round 6
            </p>
            <h4
              class="c9"
            >
              7.5
            </h4>
          </figcaption>
        </figure>
      </article>
    </li>
    <li>
      <article
        class="c3"
      >
        <figure
          class="c4"
        >
          <img
            alt="poster da série Round 6"
            class="c5"
            src="https://image.tmdb.org/t/p/original/6gcHdboppvplmBWxvROc96NJnmm.jpg"
          />
          <figcaption
            class="c6 c7"
          >
            <p
              class="c8"
            >
              Round 6
            </p>
            <h4
              class="c9"
            >
              7.5
            </h4>
          </figcaption>
        </figure>
      </article>
    </li>
    <li>
      <article
        class="c3"
      >
        <figure
          class="c4"
        >
          <img
            alt="poster da série Round 6"
            class="c5"
            src="https://image.tmdb.org/t/p/original/6gcHdboppvplmBWxvROc96NJnmm.jpg"
          />
          <figcaption
            class="c6 c7"
          >
            <p
              class="c8"
            >
              Round 6
            </p>
            <h4
              class="c9"
            >
              7.5
            </h4>
          </figcaption>
        </figure>
      </article>
    </li>
    <li>
      <article
        class="c3"
      >
        <figure
          class="c4"
        >
          <img
            alt="poster da série Round 6"
            class="c5"
            src="https://image.tmdb.org/t/p/original/6gcHdboppvplmBWxvROc96NJnmm.jpg"
          />
          <figcaption
            class="c6 c7"
          >
            <p
              class="c8"
            >
              Round 6
            </p>
            <h4
              class="c9"
            >
              7.5
            </h4>
          </figcaption>
        </figure>
      </article>
    </li>
    <li>
      <article
        class="c3"
      >
        <figure
          class="c4"
        >
          <img
            alt="poster da série Round 6"
            class="c5"
            src="https://image.tmdb.org/t/p/original/6gcHdboppvplmBWxvROc96NJnmm.jpg"
          />
          <figcaption
            class="c6 c7"
          >
            <p
              class="c8"
            >
              Round 6
            </p>
            <h4
              class="c9"
            >
              7.5
            </h4>
          </figcaption>
        </figure>
      </article>
    </li>
  </ul>
</section>
`;
